### PR TITLE
WooCommerce: Track events for dashboard buttons, payment methods, and store link.

### PR DESCRIPTION
**Pull request by justinshreve**
See #15316. This PR adds tracking for the following events mentioned in the issue:

* Number of clicks on 'store' nav item
* Each button click on dashboard
* Payment Methods Used (Enabled & Disabled Counts)

This does not do any sort of tracking for orders or products yet.

I did not implement the following:
* `number who fill out the pre-egg store location` cc @allendav can you add this as you complete that section?
* `shipping methods used` cc @marcinbot  can you add that as you complete this section?

To Test:

* Go to `https://developer.wordpress.com/docs/api/console/` and make a post request to `/sites/:site/calypso-preferences/woocommerce`, setting each field to 0. See p6q8Tx-Cn-p2 for more information.
* Go to http://calypso.localhost:3000/store/:site
* Open your dev console and enter `localStorage.setItem('debug', 'calypso:analytics:*');` [See Analytics Docs](https://github.com/Automattic/wp-calypso/tree/master/client/lib/analytics#tracks-api).
* You should see the `Howdy! Let's set up your store & start selling` page.
* Click the various action buttons, and confirm that you see messages in the console for the action. Example: `calypso:analytics:tracks Record event`.
* Click "I'm finished setting up". This should also record an event.
* Test the buttons on this version of the dashboard to make sure these record events too.
* Go to `Settings > Payments`.
* Enable and disable payment methods and see that events are recorded for these actions.
* Click the site selector arrow so that you go to the stats / normal my sites view for your site.
* Click the `Store (BETA)` link, and verify an event is recorded for that link.

@kellychoffman

### CODE REVIEW

**Comment by allendav** [comment]
> number who fill out the pre-egg store location cc @allendav can you add this as you complete that section?

hey @justinshreve - yes, I can do that in "part 3" of pre-egg

File: client/extensions/woocommerce/app/settings/payments/payment-method-item.js [hunk 1]
```diff
@@ -81,6 +82,16 @@ class PaymentMethodItem extends Component {
 			enabled
 		);
 
+		if ( enabled ) {
+			analytics.tracks.recordEvent( 'calypso_woocommerce_payment_method_enabled', {
+				paymentMethod: method.id,
```
Comment (allendav) [inline client/extensions/woocommerce/app/settings/payments/payment-method-item.js]: In this context, is id a number or a string like 'paypal' ? (Hoping it is the latter)

File: client/extensions/woocommerce/components/basic-widget/index.js [hunk 2]
```diff
@@ -3,22 +3,24 @@
...
 import React, { PropTypes } from 'react';
+import { noop } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
 
-const BasicWidget = ( { buttonLabel, buttonLink, children, className, title } ) => {
+const BasicWidget = ( { buttonLabel, buttonLink, buttonClick, children, className, title } ) => {
```
Comment (allendav) [inline client/extensions/woocommerce/components/basic-widget/index.js]: Would you mind renaming buttonClick to onButtonClick to make it clearer it is a function

File: client/my-sites/sidebar/sidebar.jsx [hunk 3]
```diff
@@ -336,6 +336,11 @@ export class MySitesSidebar extends Component {
 		);
 	}
 
+	trackStoreClick = () => {
+		analytics.tracks.recordEvent( 'calypso_woocommerce_store_nav_item_click' );
```
Comment (allendav) [inline client/my-sites/sidebar/sidebar.jsx]: But we won't know which thing they clicked on, will we? Could we add that?

File: client/extensions/woocommerce/app/settings/payments/payment-method-item.js (hunk 1 above)
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/payments/payment-method-item.js]: String!

File: client/my-sites/sidebar/sidebar.jsx (hunk 3 above)
Comment (justinshreve) [inline client/my-sites/sidebar/sidebar.jsx]: This one tracks just the (`Store (BETA)`) link by itself, not any of the other nav items.

Comment (justinshreve) [inline client/my-sites/sidebar/sidebar.jsx]: I'll make sure each track event in MC gets a description so we will know what these mean when viewing them.

**Pull request merged**